tests/PySys/software-management-end-to-end/environment_sm_management.py

```diff
@@ -0,0 +1,267 @@
+"""
+This environment provides an interface to software management features
+through the C8y REST API.
+With these we can emulate a user doing operations in the C8y UI.
+They are rather slow as they use the complete chain from end to end.
+
+WARNING: Handle with care!!!
+The C8YDEVICEID will handle on which this test will install and remove packages.
+
+Theese tests are disabled by default.
+To run them call:
+
+pysys.py run 'sm-apt*' -XmyPlatform='specialcontainer'
+
+"""
+
+import base64
+import time
+
+import json
+import requests
+
+import pysys
+from pysys.basetest import BaseTest
+
+
+def is_timezone_aware(stamp):
+    """determine if object is timezone aware or naive
+    See also: https://docs.python.org/3/library/datetime.html?highlight=tzinfo#determining-if-an-object-is-aware-or-naive
+    """
+    return stamp.tzinfo is not None and stamp.tzinfo.utcoffset(stamp) is not None
+
+
+class SmManagement(BaseTest):
+    """Base class for software management tests"""
+
+    # Static class member that can be overriden by a command line argument
+    # E.g.:
+    # pysys.py run 'sm-apt*' -XmyPlatform='specialcontainer'
+    myPlatform = None
+
+    tenant_url = "thin-edge-io.eu-latest"
+
+    def setup(self):
+        """Setup Environment"""
+
+        if self.myPlatform != "specialcontainer":
+            self.skipTest("Testing the apt plugin is not supported on this platform")
+
+
+        tenant = self.project.tenant
+        user = self.project.username
+        password = self.project.c8ypass
+
+        # Place to save the id of the operation that we started.
+        # This is suitable for one operation and not for multiple ones running
+        # at the same time.
+        self.operation_id = None
+
+        auth = bytes(f"{tenant}/{user}:{password}", "utf-8")
+        self.header = {
+            b"Authorization": b"Basic " + base64.b64encode(auth),
+            b"content-type": b"application/json",
+            b"Accept": b"application/json",
+        }
+
+        # Make sure we have no last operations pending or executing
+        self.wait_until_end()
+
+    def trigger_action(self, package_name, package_id, version, url, action):
+        """Trigger a installation or deinstallation of a package.
+        package_id is the id that is automatically assigned by C8y.
+        """
+
+        self.trigger_action_json(
+            [
+                {
+                    "name": package_name,
+                    "id": package_id,
+                    "version": version,
+                    "url": url,
+                    "action": action,
+                }
+            ]
+        )
+
+    def trigger_action_json(self, json_content):
+        """Take an actions description that is then forwarded to c8y.
+        So far, no checks are done on the json_content.
+        """
+
+        url = f"https://{self.tenant_url}.cumulocity.com/devicecontrol/operations"
```

Comment:
To avoid that we need to update the workflow I added the parameter as static member.
For now you can select your own url with:
pysys.py run 'sm-apt*' -XmyPlatform='specialcontainer' -Xtenant_url='thin-edge-io.eu-latest.cumulocity.com'